src/widgets/widget_ext.rs

```diff
@@ -0,0 +1,50 @@
+//! A module that provides an extension trait for widgets that provides methods that are useful for
+//! debugging.
+
+use super::ansi_string_buffer::AnsiStringBuffer;
+use crate::prelude::*;
+
+/// An extension trait for widgets that provides methods that are useful for debugging.
+#[stability::unstable(
+    feature = "widget-ext",
+    issue = "https://github.com/ratatui-org/ratatui/issues/1045"
+)]
+pub trait WidgetExt {
```

Comment:
I agree with your comment about the naming. In the off chance we want to use more extensions, this should be named `ToAnsiStringWidgetExt` or something to that effect.

Reply:
Either this trait should allow for more things on the same trait (in the future), then the name is fine, or it should be specific to this.
I also dislike the Ext ending here.
`AnsiStringPrintable` would be a trait name that represents what it does.

Reply:
I think Ext is *probably* the right suffix for this pattern if it's not a more generic trait. https://rust-lang.github.io/rfcs/0445-extension-trait-conventions.html. It's used in tokio / futures streams (StreamExt), tracing-subscriber (a bunch of times https://docs.rs/tracing-subscriber/latest/tracing_subscriber/?search=ext) and various other crates use similar naming. I mostly wasn't sure if adding something before Ext made sense.

> The proposed convention is, first of all, to (1) prefer adding default methods to existing traits or (2) prefer generically useful traits to extension traits whenever feasible.

Adding default methods doesn't seem like the right idea as converting to ANSI doesn't seem like it's something everyone would need on a widget, and it's something I'd prefer not to confuse newer users with if possible (hence adding as an extension trait / generically useful trait seems right). Adding a default method would however allow each widget to define their own ANSI representation (though I can't think of a useful need for this).

I'm not sure that given that the method needs a certain size to render, whether this trait has a generic usefulness. I don't see that there's really any need for this trait other than printing out widgets (I could be wrong there).

Reply:
Other objects not being Widgets could implement that trait too. Buffer for example.

Reply:
I was also thinking `Ext` was the right suffix since I had seen it in several other places, but maybe it is more commonly used when extending external traits or standard library traits? I like `AnsiStringPrintable` too. 

If it has just one method `.to_ansi_string`, why not `trait ToAnsiString {}`?

Reply:
> Other objects not being Widgets could implement that trait too. Buffer for example.

The bit I struggle with is working out what the purpose of passing in the width and height would be if implemented on Buffer? Would `trait RenderToAnsiString { fn render_to_ansi_string() }` make sense?